tempesta_fw/t/functional/regression/test_health_monitor.py

```diff
@@ -0,0 +1,227 @@
+"""
+Tests for health monitoring functionality.
+"""
+
+from __future__ import print_function
+import re
+import copy
+from testers import functional
+from helpers import deproxy, chains, tempesta
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2017 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+
+def make_response(st_code, expected=True):
+    resp_headers = [
+        'Content-Length: 0',
+        'Connection: keep-alive'
+    ]
+    if expected:
+        resp_headers += [
+            'Server: Tempesta FW/%s' % tempesta.version(),
+            'Via: 1.1 tempesta_fw (Tempesta FW %s)' % tempesta.version()
+        ]
+    else:
+        resp_headers += ['Server: Deproxy Server']
+    response = deproxy.Response.create(
+        status=st_code,
+        headers=resp_headers,
+        date=deproxy.HttpMessage.date_time_string()
+    )
+    return response
+
+def make_502_expected():
+    response = deproxy.Response.create(
+        status=502,
+        headers=['Content-Length: 0', 'Connection: keep-alive'],
+        date=deproxy.HttpMessage.date_time_string()
+    )
+    return response
+
+
+class TestHealthMonitor(functional.FunctionalTest):
```

Comment:
Please comment in the code each test case from https://github.com/tempesta-tech/tempesta/pull/877#discussion_r159697039

Reply:
Done.